test/e2e/example-server/main.go

```diff
@@ -49,6 +49,7 @@ func init() {
 
 func ServerHTTP(writer http.ResponseWriter, request *http.Request) {
 	time.Sleep(time.Duration(500) * time.Millisecond)
+	go2sky.PutCorrelation(request.Context(), "MIDDLE_KEY", "go2sky")
```

Comment:
I think you need to read this context and add it into tag, then, you could verify.

Reply:
This E2E is running at the skywalking main repo, and execute it in the following way:
`consumer -> go2sky -> provider`.
1. `consumer` and `provider` both are java applications.
2. all processes add the correlation key.
3. the `provider` return all the correlation result, so we will know all the correlation keys and values.
4. follow these steps, we will test the go2sky has received, save, and add correlation context ability. 

Do you mean to add the tag at the tracing context? such as add tags from consumer keys?

Reply:
My point is, you added one more correlation key, but no one verifies whether it is working.

Reply:
It will verify at the main repo. The `provider` will get all of the correlation values, and E2E verify the result of the value is the same.

Reply:
This should be a local e2e, why the verification happens in the main repo? What do I miss?

Reply:
This `main.go` is running at the main repo E2E only. The project itself has no place to run.

Reply:
OK, that is new to me.